src/Adapter/Module/ModuleDataProvider.php

```diff
@@ -370,4 +368,18 @@ private function getDeviceStatus($name)
 
         return false;
     }
+
+    private function isModuleActive(int $id, array $id_shops): ?array
+    {
+        $result = Db::getInstance()->getRow('SELECT m.`id_module` as `active`, ms.`id_module` as `shop_active`, ms.`enable_device` as `enable_device`
+            FROM `' . _DB_PREFIX_ . 'module` m
+            LEFT JOIN `' . _DB_PREFIX_ . 'module_shop` ms ON m.`id_module` = ms.`id_module`
+            WHERE m.`id_module` = ' . $id . '
+            AND ms.`id_shop` IN (' . implode(',', $id_shops) . ')');
+        if ($result) {
+            return $result;
```

Comment:
It's a bit weird you return an associative array of value but never check their content, the returned type is an array and is cast as a boolean Shouldn't the method return a boolean instead?

Reply:
Something like:
```suggestion
            return !empty($result['enable_device']) && !empty($result['active']);
```